+nosnoc/+discrete_problem/stewart.m

```diff
@@ -0,0 +1,771 @@
+                strcmp(obj.opts.step_equilibration,'linear_relaxed'))
+                obj.w.B_max(1:opts.N_stages,2:opts.N_finite_elements) = {{'B_max', dims.n_lambda},-inf,inf};
+                obj.w.pi_theta(1:opts.N_stages,2:opts.N_finite_elements(1)) = {{'pi_theta', dims.n_theta},-inf,inf};
+                obj.w.pi_lambda(1:opts.N_stages,2:opts.N_finite_elements(1)) = {{'pi_lambda', dims.n_lambda},-inf,inf};
+                obj.w.lambda_theta(1:opts.N_stages,2:opts.N_finite_elements(1)) = {{'lambda_theta', dims.n_theta},0,inf};
+                obj.w.lambda_lambda(1:opts.N_stages,2:opts.N_finite_elements(1)) = {{'lambda_lambda', dims.n_lambda},0,inf};
+                obj.w.eta(1:opts.N_stages,2:opts.N_finite_elements(1)) = {{'eta', dims.n_lambda},0,inf};
+                obj.w.nu(1:opts.N_stages,2:opts.N_finite_elements(1)) = {{'nu', 1},0,inf};
+            end
+
+            % For c_n ~= 1 case
+            rbp = ~opts.right_boundary_point_explicit;
+            % 3d vars
+            obj.w.x(0,0,opts.n_s) = {{['x_0'], dims.n_x}, model.x0, model.x0, model.x0};
+            if (opts.irk_representation == IrkRepresentation.integral ||...
+                        opts.irk_representation == IrkRepresentation.differential_lift_x)
+                obj.w.x(1:opts.N_stages,1:opts.N_finite_elements(1),1:(opts.n_s+rbp)) = {{'x', dims.n_x}, model.lbx, model.ubx, model.x0};
+            else
+                obj.w.x(1:opts.N_stages,1:opts.N_finite_elements(1),opts.n_s) = {{'x', dims.n_x}, model.lbx, model.ubx, model.x0};
+            end
+            if (opts.irk_representation == IrkRepresentation.differential ||...
+                opts.irk_representation == IrkRepresentation.differential_lift_x)
+                obj.w.v(1:opts.N_stages,1:opts.N_finite_elements(1),1:opts.n_s) = {{'v', dims.n_x}};
+            end
+            obj.w.z(0,0,opts.n_s) = {{'z', dims.n_z}, model.lbz, model.ubz, model.z0};
+            obj.w.z(1:opts.N_stages,1:opts.N_finite_elements(1),(opts.n_s+rbp)) = {{'z', dims.n_z}, model.lbz, model.ubz, model.z0};
+            obj.w.lambda(0,0,opts.n_s) = {{['lambda'], dims.n_lambda},0,inf};
+            obj.w.lambda(1:opts.N_stages,1:opts.N_finite_elements(1),1:(opts.n_s+rbp)) = {{'lambda', dims.n_lambda},0, inf};
+            obj.w.theta(1:opts.N_stages,1:opts.N_finite_elements(1),1:(opts.n_s)) = {{'theta', dims.n_theta},0, 1};
+            obj.w.mu(0,0,opts.n_s) = {{'mu', dims.n_mu},0,inf};
+            obj.w.mu(1:opts.N_stages,1:opts.N_finite_elements(1),1:(opts.n_s+rbp)) = {{'mu', dims.n_mu},0,inf};
+
+            % Handle x_box settings
+            if ~opts.x_box_at_stg && opts.irk_representation ~= IrkRepresentation.differential
+                obj.w.x(1:opts.N_stages,1:opts.N_finite_elements(1),1:(opts.n_s+rbp-1)).lb = -inf*ones(dims.n_x, 1);
+                obj.w.x(1:opts.N_stages,1:opts.N_finite_elements(1),1:(opts.n_s+rbp-1)).ub = inf*ones(dims.n_x, 1);
+            end
+
+            if ~opts.x_box_at_fe
```

Comment:
Maybe we should have enums how fine to discretize path and box constraints, instead of separate options with true false for every possiblity

Reply:
Agreed. Also we need to think about what namespace the Enums should live in/should we keep them in the global namespace due to matlab's horrible autocomplete.